inference-engine/thirdparty/clDNN/src/network.cpp

```diff
@@ -189,16 +196,18 @@ static void dump(memory::ptr mem, stream& stream, std::ofstream& file_stream) {
 
     mem_lock<T> lock(mem, stream);
     auto mem_ptr = lock.data();
+    auto x_pitch = get_x_pitch(mem->get_layout());
 
     for (cldnn::tensor::value_type g = 0; g < size.group[0]; ++g) {
         for (cldnn::tensor::value_type b = 0; b < size.batch[0]; ++b) {
             for (cldnn::tensor::value_type f = 0; f < size.feature[0]; ++f) {
                 for (cldnn::tensor::value_type w = 0; w < size.spatial[3]; ++w) {
                     for (cldnn::tensor::value_type z = 0; z < size.spatial[2]; ++z) {
                         for (cldnn::tensor::value_type y = 0; y < size.spatial[1]; ++y) {
-                            for (cldnn::tensor::value_type x = 0; x < size.spatial[0]; ++x) {
-                                cldnn::tensor t(cldnn::group(g), cldnn::batch(b), cldnn::feature(f), cldnn::spatial(x, y, z, w));
-                                size_t input_it = mem->get_layout().get_linear_offset(t);
+                            cldnn::tensor t(cldnn::group(g), cldnn::batch(b), cldnn::feature(f), cldnn::spatial(0, y, z, w));
+                            size_t input_it = mem->get_layout().get_linear_offset(t);
+
+                            for (cldnn::tensor::value_type x = 0; x < size.spatial[0]; ++x, input_it += x_pitch) {
                                 file_stream << std::fixed << std::setprecision(6) << convert_element(mem_ptr[input_it]) << std::endl;
```

Comment:
I think we can store to std::stringstream directly and the end of dumping push whole stringstream.str() to file stream. According to my tests, we can get a speed up up to 4 times.

Reply:
Thanks for the idea. I confirmed that it has significant performance gain!

Reply:
BTW, if we talk about performance of the dump feature, then we may want to support saving to binary file instead of text one. It'll be much faster, but we'll need to implement a script to compare binary files

Reply:
Interesting point. For now, I'm happy with this performance gain ;) We may work further in the future to shorten the time more.